py/server/deephaven/table.py

```diff
@@ -47,12 +48,16 @@
 
 # Dynamic Query Scope
 _JExecutionContext = jpy.get_type("io.deephaven.engine.context.ExecutionContext")
-_JQueryScope = jpy.get_type("io.deephaven.engine.context.QueryScope")
 _JUnsynchronizedScriptSessionQueryScope = jpy.get_type(
     "io.deephaven.engine.util.AbstractScriptSession$UnsynchronizedScriptSessionQueryScope")
 _JPythonScriptSession = jpy.get_type("io.deephaven.integrations.python.PythonDeephavenSession")
-_j_script_session = jpy.cast(_JExecutionContext.getContext().getQueryScope(), _JUnsynchronizedScriptSessionQueryScope).scriptSession()
-_j_py_script_session = jpy.cast(_j_script_session, _JPythonScriptSession)
+
+
+@functools.lru_cache(maxsize=None)
```

Comment:
Note: I've chosen to cache this method call to preserve the "speed" of `_query_scope_ctx()`. We may decide that it's reasonable to do this invocation every time instead of caching the value - in which case, I'm happy to remove the cache altogether. Thoughts?

Reply:
I think the execution context work enabled us to have multiple script sessions if we want to. Does this effectively break that?

Reply:
there is another issue with this lazy eval approach, if the very first table operation happens in non-main thread, then we will be hitting the poisoned execution context.